src/transfers/transferTransactionsFactory.ts

```diff
@@ -20,50 +19,18 @@ interface IConfig {
     gasLimitMultiESDTNFTTransfer: bigint;
 }
 
-interface IGasEstimator {
-    forEGLDTransfer(dataLength: number): number;
-    forESDTTransfer(dataLength: number): number;
-    forESDTNFTTransfer(dataLength: number): number;
-    forMultiESDTNFTTransfer(dataLength: number, numTransfers: number): number;
-}
-
 /**
  * Use this class to create transactions for native token transfers (EGLD) or custom tokens transfers (ESDT/NTF/MetaESDT).
  */
 export class TransferTransactionsFactory {
     private readonly config?: IConfig;
     private readonly tokenTransfersDataBuilder?: TokenTransfersDataBuilder;
     private readonly tokenComputer?: TokenComputer;
-    private readonly gasEstimator?: IGasEstimator;
 
-    /**
-     * Should be instantiated using `Config`.
-     * Instantiating this class using GasEstimator represents the legacy version of this class.
-     * The legacy version contains methods like `createEGLDTransfer`, `createESDTTransfer`, `createESDTNFTTransfer` and `createMultiESDTNFTTransfer`.
-     * This was done in order to minimize breaking changes in client code.
-     */
-    constructor(options: IGasEstimator | { config: IConfig }) {
-        if (this.isGasEstimator(options)) {
-            this.gasEstimator = options;
-        } else {
-            this.config = options.config;
-            this.tokenComputer = new TokenComputer();
-            this.tokenTransfersDataBuilder = new TokenTransfersDataBuilder();
-        }
-    }
-
-    private isGasEstimator(options: any): options is IGasEstimator {
-        return (
-            typeof options === "object" &&
-            typeof options.forEGLDTransfer === "function" &&
-            typeof options.forESDTTransfer === "function" &&
-            typeof options.forESDTNFTTransfer === "function" &&
-            typeof options.forMultiESDTNFTTransfer === "function"
-        );
-    }
-
-    private isGasEstimatorDefined(): boolean {
-        return this.gasEstimator !== undefined;
+    constructor(options: { config: IConfig }) {
+        this.config = options.config;
+        this.tokenComputer = new TokenComputer();
+        this.tokenTransfersDataBuilder = new TokenTransfersDataBuilder();
     }
 
     private ensureConfigIsDefined() {
```

Comment:
I think this can be removed since now the only way to use the factory is to instantiate it with a config